src/Popover.tsx

```diff
@@ -596,15 +614,27 @@ class AdaptivePopover extends Component<AdaptivePopoverProps, AdaptivePopoverSta
         onOpenStart={() => {
           if (onOpenStart) onOpenStart();
           this.debug('Setting up keyboard listeners');
-          Keyboard.addListener('keyboardDidShow', this.keyboardDidShow);
-          Keyboard.addListener('keyboardDidHide', this.keyboardDidHide);
+          this.setState({ keyboardDidShowSubscription: Keyboard.addListener('keyboardDidShow', this.keyboardDidShow) });
+          this.setState({ keyboardDidHideSubscription: Keyboard.addListener('keyboardDidHide', this.keyboardDidHide) });
           this.displayAreaStore = this.getDisplayArea();
         }}
         onCloseStart={() => {
           if (onCloseStart) onCloseStart();
           this.debug('Tearing down keyboard listeners');
-          Keyboard.removeListener('keyboardDidShow', this.keyboardDidShow);
-          Keyboard.removeListener('keyboardDidHide', this.keyboardDidHide);
+          const { keyboardDidShowSubscription } = this.state;
+          // Typescript is struggling to identify `keyboardDidShowSubscription` can't be null
+          if (keyboardDidShowSubscription !== null) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            (keyboardDidShowSubscription! as EmitterSubscription).remove();
```

Comment:
You can try using `if (keyboardDidShowNotification)` instead of `if (keyboardDidShowNotification !== null)`, maybe TypeScript will like that better.  You could also try updating TypeScript, maybe its a bug that was fixed.

Reply:
I just pulled your fork and ran `yarn tsc`, and it had to problem with that line as if.  If you remove the comments we should be good

Reply:
Just removed the comments, good catch.